vmaas/errata.go

```diff
@@ -0,0 +1,101 @@
+package vmaas
+
+import (
+	"slices"
+
+	"github.com/pkg/errors"
+	"github.com/redhatinsights/vmaas-lib/vmaas/utils"
+)
+
+type ErrataDetails map[string]ErratumDetail
+
+type Errata struct {
+	ErrataList ErrataDetails `json:"errata_list"`
+	Type       []string      `json:"type"`
+	Severity   StringSlice   `json:"severity,omitempty"`
+	LastChange string        `json:"last_change"`
+	utils.PaginationDetails
+}
+
+func filterInputErrata(c *Cache, errata []string, req *ErrataRequest) []string {
+	isDuplicate := make(map[string]bool, len(errata))
+	filteredErrata := make([]string, 0, len(errata))
+	for _, erratum := range errata {
+		if erratum == "" || isDuplicate[erratum] {
+			continue
+		}
+		erratumDetail, found := c.ErratumDetails[erratum]
+		if !found {
+			continue
+		}
+
+		if !req.ThirdParty && erratumDetail.ThirdParty {
+			continue
+		}
+
+		if req.ModifiedSince != nil {
+			if erratumDetail.Issued != nil && erratumDetail.Issued.Before(*req.ModifiedSince) {
+				continue
+			}
+			if erratumDetail.Updated != nil && erratumDetail.Updated.Before(*req.ModifiedSince) {
+				continue
+			}
+		}
+
+		if req.Type != nil && !slices.Contains(req.Type, erratumDetail.Type) {
+			continue
+		}
+		if req.Severity != nil && !slices.Contains(req.Severity, erratumDetail.Severity) {
+			continue
+		}
+
+		filteredErrata = append(filteredErrata, erratum)
+		isDuplicate[erratum] = true
+	}
+	return filteredErrata
+}
+
+func (c *Cache) loadErratumReleaseVersions(erratumID ErratumID) []string {
+	erratumRepos := c.ErratumID2RepoIDs[erratumID]
+	releaseVers := make([]string, 0, len(erratumRepos))
+	for repoID := range erratumRepos {
+		repoDetail := c.RepoDetails[repoID]
+		releaseVer := repoDetail.Releasever
+		if releaseVer != "" {
+			releaseVers = append(releaseVers, releaseVer)
+		}
+	}
+	return releaseVers
+}
+
+func (c *Cache) loadErrataDetails(errata []string) ErrataDetails {
+	errataDetails := make(ErrataDetails, len(errata))
+	for _, erratum := range errata {
+		erratumDetail := c.ErratumDetails[erratum]
+		binPackages, sourcePackages := c.packageIDs2Nevras(erratumDetail.PkgIDs)
+		erratumDetail.PackageList = binPackages
+		erratumDetail.SourcePackageList = sourcePackages
+		erratumDetail.ReleaseVersions = c.loadErratumReleaseVersions(erratumDetail.ID)
+		errataDetails[erratum] = erratumDetail
+	}
+	return errataDetails
+}
+
+func (req *ErrataRequest) errata(c *Cache) (*Errata, error) { // TODO: implement opts
+	if len(req.ErrataList) == 0 {
+		return nil, errors.New("errata_list must contain at least one item")
+	}
+
+	errata := utils.TryExpandRegexPattern(req.ErrataList, c.ErratumDetails)
+	errata = filterInputErrata(c, errata, req)
+	errata, paginationDetails := utils.Paginate(errata, req.PageNumber, req.PageSize)
```

Comment:
errata slice needs to be sorted before pagination, we are getting different response for the same request e.g. with page_size=10